commandapi-core/src/main/java/dev/jorel/commandapi/executors/AbstractExecutionInfo.java

```diff
@@ -0,0 +1,27 @@
+package dev.jorel.commandapi.executors;
+
+import dev.jorel.commandapi.commandsenders.AbstractCommandSender;
+
+/**
+ * This interface represents an AbstractExecutionInfo for a command. It provides the sender of a command, as well as it's arguments
+ *
+ * @param <Sender> The type of the sender of a command this AbstractExecutionInfo belongs to
+ */
+public interface AbstractExecutionInfo<Sender, WrapperType extends AbstractCommandSender<? extends Sender>> {
+
+	/**
+	 * @return The sender of this command
+	 */
+	Sender sender();
+
+	/**
+	 * @return The wrapper type of this command
+	 */
+	WrapperType senderWrapper();
```

Comment:
This is exposed to the end user. A comment stating that this is only used internally, and you should be using `sender()` instead would be nice.